core/src/main/java/org/mule/runtime/core/internal/util/JarUtils.java

```diff
@@ -58,15 +58,20 @@ private JarUtils() {
 
   /**
    * Loads the content of a file within a jar into a byte array.
-   * 
+   *
    * @param jarFile  the jar file
    * @param filePath the path to the file within the jar file
    * @return the content of the file as byte array or empty if the file does not exists within the jar file.
    * @throws IOException if there was a problem reading from the jar file.
    */
   public static Optional<byte[]> loadFileContentFrom(File jarFile, String filePath) throws IOException {
     URL jsonDescriptorUrl = getUrlWithinJar(jarFile, filePath);
-    JarURLConnection jarConnection = (JarURLConnection) jsonDescriptorUrl.openConnection();
+    /*
+     * A specific implementation of JarURLConnection is required to read jar content because not all implementations support ways
+     * to disable connection caching. Disabling connection caching is necessary to avoid file descriptor leaks.
+     */
+    JarURLConnection jarConnection =
```

Comment:
why not move code using these classes to a new module so the impact of changing the comiler configuration is contained there?